### DISCUSSION TITLE
`node_color=nx.get_node_attributes(G, "color").values()` works, while `node_size=nx.get_node_attributes(G "size")` won't

### DISCUSSION THREAD

**Discussion by germannp** [networkx/networkx#5417] [Ideas]
Hello

I feel that coloring nodes or scaling nodes by an attribute should work the same way, however `node_size` won't work with `dict_values` and requires e.g. a `list` instead:

```Python
G = nx.barbell_graph(5, 2)

for node in G.nodes:
    G.nodes[node]["color"] = np.random.choice([f"C{i}" for i in range(4)])

for node in G.nodes:
    G.nodes[node]["size"] = np.random.rand() * 2000

nx.draw(
    G,
    node_color=nx.get_node_attributes(G, "color").values(),
    node_size=list(nx.get_node_attributes(G, "size").values()),
)
```
Also the resulting error without the `list()` was not intuitive to me:
```
ValueError: s must be a scalar, or float array-like with the same size as x and y
```

Cheers,
Philipp

**Comment by rossbar** [comment]
Yeah I suspect there are a lot of kwargs that are not as strict about input types as they claim to be. In this case, `node_color` *shouldn't* work (at least according to the parameter type in the docstring) but does likely due to some internal conversion from `dict_values` under the hood. `dict_values` objects are not array-like, so I suspect the unhelpful exception message is due to the fact that there is an under-the-hood attempt to convert the dict_values directly to an array without first converting to a sequence, which results in an object array with a single element.

Generally, the current solution is: make sure that the parameter input types are supported. In principle static typing would help here, at least for non-interactive code. Otherwise the solution is to add more `isinstance` checks to protect against `dict_values` and other non-array-like inputs for the parameters that support array-likes.
